packages/image_picker/image_picker_ios/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -575,47 +605,42 @@ - (void)saveImageWithPickerInfo:(NSDictionary *)info
   NSString *savedPath = [FLTImagePickerPhotoAssetUtil saveImageWithPickerInfo:info
                                                                         image:image
                                                                  imageQuality:imageQuality];
-  [self handleSavedPathList:@[ savedPath ]];
+  [self returnSavedPathList:@[ savedPath ]];
 }
 
 /**
- * Applies NSMutableArray on the FLutterResult.
- *
- * NSString must be returned by FlutterResult if the single image
- * mode is active. It is checked by maxImagesAllowed and
- * returns the first object of the pathlist.
+ * Validates the provided paths list, then returns it as the result of the original method call,
+ * clearing the in-progress call state.
  *
- * NSMutableArray must be returned by FlutterResult if the multi-image
- * mode is active. After the pathlist count is checked then it returns
- * the pathlist.
- *
- * @param pathList that should be applied to FlutterResult.
+ * @param pathList The paths to return. nil indicates a cancelled operation.
  */
-- (void)handleSavedPathList:(NSArray *)pathList {
-  if (!self.result) {
+- (void)returnSavedPathList:(nullable NSArray *)pathList {
```

Comment:
nits: The method name suggests it should return the list. Is there a better way to rename it so that it is clearer that this method is sending the result to method channel. (I can't think of any though)
If no better name is found, maybe update the doc so it is more explicit:

"then returns it in `callContext.result` of the original method channel call"

Reply:
Renamed both of the response methods to `sendCallResultWith*`, and also updated the comments per your suggestion, so they should be much more explicit about what they are doing.

(Long term, the real solution is to get rid of these methods.)